handlers/src/main/java/com/nextdoor/bender/handler/kinesis/KinesisEventIterator.java

```diff
@@ -54,7 +69,19 @@ public boolean hasNext() {
 
   @Override
   public InternalEvent next() {
-    return new KinesisInternalEvent(this.iterator.next(), this.context, this.shardId);
+    KinesisEventRecord nextRecord = this.iterator.next();
+    if (assumeKinesisDataIsGzipped) {
+      ByteBuffer gzip = nextRecord.getKinesis().getData();
+      try (GZIPInputStream gzipInputStream = new GZIPInputStream(new ByteArrayInputStream(gzip.array()))) {
```

Comment:
Let's pull in `ByteBufferInputStream` into the project and use it here. https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/util/ByteBufferInputStream.java

Reply:
Added to bender common module for usage.